packages/jellyfish-api-core/__tests__/category/loan/listAuctons.test.ts

```diff
@@ -0,0 +1,521 @@
+          liquidationPenalty: result?.liquidationPenalty,
+          vaultId: vaultId1
+        }
+      )
+    })
+  })
+
+  describe('listAuctions with pagination', () => {
+    let auction1: AuctionDetail
+    let auction2: AuctionDetail
+    let auction3: AuctionDetail
+    let auction4: AuctionDetail
+
+    beforeAll(async () => {
+      [auction1, auction2, auction3, auction4] = await testing.rpc.loan.listAuctions()
+    })
+
+    it('should listAuctions with limit only', async () => {
+      // List auctions with limit = size (When pass in limit only, including_start always = true)
+      const pageLimit4 = await testing.rpc.loan.listAuctions({ limit: 4 })
+      expect(pageLimit4.length).toStrictEqual(4)
+      expect(pageLimit4[0].vaultId).toStrictEqual(auction1.vaultId)
+      expect(pageLimit4[1].vaultId).toStrictEqual(auction2.vaultId)
+      expect(pageLimit4[2].vaultId).toStrictEqual(auction3.vaultId)
+      expect(pageLimit4[3].vaultId).toStrictEqual(auction4.vaultId)
+
+      // List auctions with limit > size
+      const pageLimit5 = await testing.rpc.loan.listAuctions({ limit: 5 })
+      expect(pageLimit5.length).toStrictEqual(4)
+      expect(pageLimit5).toStrictEqual(pageLimit4)
+    })
+
+    it('should listAuctions with height', async () => {
+      // List for liquidation height of first vault
+      // including_start = true
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { height: auction1.liquidationHeight }, including_start: true }
+        )
+        expect(page.length).toStrictEqual(4)
+        expect(page[0].vaultId).toStrictEqual(auction1.vaultId)
+        expect(page[1].vaultId).toStrictEqual(auction2.vaultId)
+        expect(page[2].vaultId).toStrictEqual(auction3.vaultId)
+        expect(page[3].vaultId).toStrictEqual(auction4.vaultId)
+      }
+      // including_start = false
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { height: auction1.liquidationHeight } } // Default = including_start = false
+        )
+        expect(page.length).toStrictEqual(3)
+        expect(page[0].vaultId).toStrictEqual(auction2.vaultId)
+        expect(page[1].vaultId).toStrictEqual(auction3.vaultId)
+        expect(page[2].vaultId).toStrictEqual(auction4.vaultId)
+      }
+
+      // List for liquidation height of forth vault
+      // including_start = true
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { height: auction4.liquidationHeight }, including_start: true }
+        )
+        expect(page.length).toStrictEqual(1)
+        expect(page[0].vaultId).toStrictEqual(auction4.vaultId)
+      }
+      // including_start = false
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { height: auction4.liquidationHeight } }
+        )
+        expect(page.length).toStrictEqual(0)
+      }
+    })
+
+    it('should listAuctions with vaultId', async () => {
```

Comment:
ListAuction with vaultId only won't work and this is not recommended. 
Added test unit here and show that the result is unpredictable.

Reply:
Added the test unit here because the return result is predictable. The result is filtered by including_start only.

Reply:
Looks like some misunderstanding here, don't see anything wrong in the RPC behavior.

All 4 listed data are having the correct vaultId right?
The including_start mean using the offset itself like (greater than vs greater than and equal to)

Reply:
pagination and filtering are two different things. Here can not filter by vault id.
<heigh, vault id> is only for pagination to give the start point in the DB. Hence should be considered together. db has the combined index <heigh, vault id>  for this.

Reply:
we can remove this test I think😛

Reply:
I think the test is correct though, it is testing the `vaultId` as offset not filter. But the response simply doesn't seems to respect the offset correctly.

full: [a, b, c, d]
start from "b" still include [a, b, c, d]

seems like a RPC bug to me

Reply:
can not use `vault id` only.
when you use that the height has the default value of 0; The index is  <height, vault ID>, not other way around as per current implementation.
```
{
        READWRITE(WrapBigEndian(height));
        READWRITE(vaultId);
    }
```
so in this case it will include all the auctions in the db

Reply:
Changed to `auction2.vaultId`
In my head, I have been questioning myself whether should I add this test or not for 10 times.

Reply:
not sure why they changed the order though😛
@Jouzo

Reply:
The logic is that you can have multiple vaults per height but not multiple heights per vault

Reply:
It was <height, vaultId> from the start. It was just accidently inverted during the refactor form std::pair to struct

Reply:
We can keep the test since it expresses the actual behaviour of the RPC. But let's change the name
```suggestion
    it('should not filter listAuctions with vaultId only', async () => {
```

Reply:
Hesitate to change, but I am going to change anyway. For the past test units, we use the words `Should not` ONLY when the exception thrown.

Reply:
Fixed

Reply:
okay, but the current name `should listAuctions with vaultId` does not suit the test content coz it implies it's working. can change to another more appropriate.